aten/src/ATen/native/cuda/UpSampleBilinear2d.cu

```diff
@@ -234,35 +381,79 @@ static void upsample_bilinear2d_backward_out_cuda_template(
       at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock, 1024);
   cudaStream_t stream = at::cuda::getCurrentCUDAStream();
 
-  AT_DISPATCH_FLOATING_TYPES_AND_HALF(
-      grad_output.scalar_type(), "upsample_bilinear2d_backward_out_frame", [&] {
-        using accscalar_t = at::acc_type<scalar_t, true>;
-
-        auto idata = grad_input.data_ptr<scalar_t>();
-        auto odata = grad_output.data_ptr<scalar_t>();
-
-        const accscalar_t rheight = area_pixel_compute_scale<accscalar_t>(
-            input_height, output_height, align_corners, scales_h);
-        const accscalar_t rwidth = area_pixel_compute_scale<accscalar_t>(
-            input_width, output_width, align_corners, scales_w);
-
-        upsample_bilinear2d_backward_out_frame<scalar_t, accscalar_t>
-            <<<cuda::ATenCeilDiv(num_kernels, static_cast<size_t>(num_threads)),
-               num_threads,
-               0,
-               stream>>>(
-                nbatch * channels,
-                input_height,
-                input_width,
-                output_height,
-                output_width,
-                rheight,
-                rwidth,
-                align_corners,
-                idata,
-                odata);
-        C10_CUDA_KERNEL_LAUNCH_CHECK();
-      });
+  if (grad_output_.sizes() == grad_input.sizes()) {
+    grad_input.copy_(grad_output_);
+    return;
+  }
+
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(grad_output_.scalar_type(), "upsample_bilinear2d_backward_out_frame", [&] {
+    if (memory_format == at::MemoryFormat::ChannelsLast && channels >= 4 && \
+          grad_input.is_contiguous(memory_format)) {
```

Comment:
So in reality this branch is never taken, because grad_input is always allocated as contiguous, and not channels last, and channels_last backward kernel is not tested?